src/plugins/workspace/server/utils.ts

```diff
@@ -113,3 +115,23 @@ export const getDataSourcesList = (client: SavedObjectsClientContract, workspace
       }
     });
 };
+
+export const checkAndSetDefaultDataSources = async (
+  uiSettingsClient: IUiSettingsClient,
+  dataSources: string[],
+  isNeededCheck: boolean
```

Comment:
Shall we separate the `checkAndSetDefaultDataSources` to `checkAndSetDefaultDataSource` and `setDefaultDataSource`? The method name with a `isNeededCheck` flag looks a little bit confused to me.

Reply:
Im open to this, but I think we could combine this logic together with some comments.

Reply:
We'd better not call different methods in create and update.

Reply:
Call with different methods is make sense to me.
For new creating workspace, we don't need to check if default data source has been set. We can just call `setDefaultDataSource` directly.
For new updating workspace, we always check if the default data source has been set. We will call `setDefaultDataSources` if default data source not set.
BTW,  There are two cases about data source been deleted:
1. All data source has been unassigned during workspace update
2. The default data source has been deleted out of workspace.
Will current implementation handle these two cases? Do we need to clear the default data source after data source been deleted?

Reply:
We have judged the lengths of array,we will only set when length is more than 0.

Reply:
Yeah. But we won't clear the old default data source if the length of array equal zero, right? For example, create workspace with 3 datasources. The default data source will be the data source 1. And then remove all the data sources in the workspace update page. Will the default data source be cleared? Is there any issues if the default data source still data source 1?

Reply:
Could you list any issues if existing?

Reply:
Thanks for your comments,that would be more safe if we do a clear operation if there is no data source left. I would add this.